carbon-design-system/carbon-for-ibm-dotcom
# feat(content-group-banner): add Content Group Banner component

### Related Ticket(s)

DDS Tickets: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5354 & https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5506
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-1736

### Description
- This PR adds the "Content Group - Banner" layout component/pattern that's needed in the new Solution page template that will be implemented in the AEM CMS.
- This component is a variant of HC's [Module 12A](https://pages.github.ibm.com/MSC-Cloud/hc_fractal/components/detail/12a--default.html) and this iteration is the first version that will be going into the MVE state of the Solution template.

### Design Spec
https://ibm.box.com/s/tqp7c0wm99ha0b7jrm0sqhq3whwdjv84
![Screen Shot 2021-04-01 at 11 10 28 AM](https://user-images.githubusercontent.com/1815714/113315076-e6583b80-92da-11eb-92bf-a23923a57953.png)


### Changelog

**New Components**

- `<dds-content-group-banner>` – a banner-like pattern that is a variant of `<dds-content-group>`, and includes a heading and link list.
- `<dds-content-section-banner>` – a Content Group Banner component assembled in a section with a left-column header.

**Removed**

- N/A

**Testing Instructions**
- Visit the `Content Group Banner` - `/?path=/story/components-content-group-banner--default`
- Ensure that the component matches with the design specs for all breakpoints.
- Visit the `Content Section Banner` - `/?path=/story/components-content-section-banner--default`
- Ensure that the card section headline renders correct and matches with the design specs for all breakpoints.
